File: src/main/java/org/visual/context/DIContext.java
package org.visual.context;

import io.avaje.inject.BeanScope;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;
import org.visual.api.ObjectDI;

@Getter
public enum DIContext implements ObjectDI {
  INSTANCE;

  private final BeanScope injector = BeanScope.builder().build();

  @Override
  public <T> @NotNull T get(Class<T> clazz) {
    return injector.get(clazz);
  }
}

File: src/main/kotlin/org/visual/context/DIContext.kt
package org.visual.context

import com.google.inject.Guice
import com.google.inject.Injector
import org.visual.module.RootModule

object DIContext {

  private val rootModule by lazy {
    RootModule()
  }

  private val injector by lazy {
    Guice.createInjector(rootModule)
  }
}